modules/admin-ui-frontend/app/scripts/shared/directives/passwordStrengthDirective.js

```diff
@@ -0,0 +1,93 @@
+/**
+ * Licensed to The Apereo Foundation under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ *
+ * The Apereo Foundation licenses this file to you under the Educational
+ * Community License, Version 2.0 (the "License"); you may not use this file
+ * except in compliance with the License. You may obtain a copy of the License
+ * at:
+ *
+ *   http://opensource.org/licenses/ecl2.txt
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ *
+ */
+
+/**
+ * @ngdoc directive
+ * @description
+ * Checks the strength of a password.
+ *
+ * @example
+ * <pw-strength ng-model="model.password"></pw-strength>
+ */
+
+'use strict';
+
+angular.module('adminNg.directives')
+    .directive('pwStrength', [function () {
+      return {
+        restrict: 'E',
+        scope: {
+          password: '=ngModel'
+        },
+
+        link: function (scope, elem, attrs, ctrl) {
+          scope.$watch('password', function (pw) {
+
+            scope.strength =
+                isValid(pw && /[A-Z]/.test(pw)) +
+                isValid(pw && /[a-z]/.test(pw)) +
+                isValid(pw && /\d/.test(pw)) +
+                isValid(pw && /(?=.*\W)/.test(pw)) +
+                isValid(pw && /^.{8,}$/.test(pw));
+
+            var bar = 'background:white; width:0%;';
+            var pw_strength = '';
+            setProgBar(scope.strength);
+
+            function isValid(rule) {
+              return rule ? 1 : 0;
+            }
+
+            function setProgBar(validRules) {
+              if (validRules == 1) {
```

Comment:
Maybe additionally check against the most commonly used bad passwords:

```suggestion
// bad passwords from https://en.wikipedia.org/wiki/List_of_the_most_common_passwords
// plus Opencast's default password
const bad_passwords = ["0", "111111", "1111111", "123", "123123", "123321",
	"1234", "12345", "123456", "1234567", "12345678", "123456789", "1234567890",
	"12345679", "123qwe", "18atcskd2w", "1q2w3e", "1q2w3e4r", "1q2w3e4r5t",
	"3rjs1la7qe", "555555", "654321", "666666", "7777777", "888888",
	"987654321", "aa12345678", "abc123", "admin", "dragon", "Dragon", "google",
	"iloveyou", "Iloveyou", "lovely", "Monkey", "mynoob", "password",
	"password1", "password123", "princess", "qwerty", "qwerty123", "qwertyuiop",
	"Qwertyuiop", "welcome", "zxcvbnm", "opencast" ]

              if (bad_passwords.indexOf(pw) >= 0) {
                  // This is one of the most commonly used (bad) passwords
                  bar = 'background:black; width:10%;';
                  pw_strength = 'Bad password';
              }
              else if (validRules == 1) {
```